src/chart-addons/stacked-legend.js

```diff
@@ -135,23 +150,36 @@ export function handleLegendLock({ locked, index = 0 }) {
   )
 
   const { encoding: { color } } = layer.getState()
-  if (typeof color.scale === "object") {
+  let redraw = false
+  if (typeof color.scale === "object") { // this if or raster-layer-heatmap-mixin.js
     if (color.legend.locked) {
-      layer.setState(setColorScaleDomain(layer.colorDomain()))
+      layer.setState(setColorScaleDomain_v2(layer.colorDomain()))
     } else {
-      layer.setState(setColorScaleDomain("auto"))
+      layer.setState(setColorScaleDomain_v2("auto"))
+      redraw = true
+    }
+  } else if (typeof color.scale === "undefined" && typeof color.domain !== "undefined") {
+    if (color.legend.locked && color.domain === "auto") {
```

Comment:
There is another property legend.open besides legend.locked, and I am not sure the difference between those two. Does legend.open used somewhere or is it redundant property from immerse? I see legend.open is assigned for Pointmap and Heat map in immerse.

Reply:
I do not know. Not familiar with the legend code. It seems like it's redundant.